src/test/kotlin/net/atos/zac/app/ontkoppeldedocumenten/OntkoppeldeDocumentenRESTServiceTest.kt

```diff
@@ -0,0 +1,202 @@
+import net.atos.zac.app.ontkoppeldedocumenten.converter.RESTOntkoppeldDocumentListParametersConverter
+import net.atos.zac.documenten.OntkoppeldeDocumentenService
+import net.atos.zac.documenten.model.OntkoppeldDocument
+import net.atos.zac.policy.PolicyService
+import net.atos.zac.policy.output.createWerklijstRechten
+import java.net.URI
+import java.util.Optional
+import java.util.UUID
+
+class OntkoppeldeDocumentenRESTServiceTest : BehaviorSpec({
+    val ontkoppeldeDocumentenService = mockk<OntkoppeldeDocumentenService>()
+    val drcClientService = mockk<DrcClientService>()
+    val zrcClientService = mockk<ZRCClientService>()
+    val ontkoppeldDocumentConverter = mockk<RESTOntkoppeldDocumentConverter>()
+    val listParametersConverter = mockk<RESTOntkoppeldDocumentListParametersConverter>()
+    val userConverter = mockk<RESTUserConverter>()
+    val policyService = mockk<PolicyService>()
+
+    val sut = OntkoppeldeDocumentenRESTService(
+        ontkoppeldeDocumentenService,
+        drcClientService,
+        zrcClientService,
+        ontkoppeldDocumentConverter,
+        listParametersConverter,
+        userConverter,
+        policyService
+    )
+
+    isolationMode = IsolationMode.InstancePerTest
+
+    beforeContainer { testCase ->
+        // only run before Given
+        if (testCase.parent == null) {
+            clearAllMocks()
+        }
+    }
+
+    Given("an id that doesn't belong to a document in the database") {
+        val id: Long = 1
+        val werklijstRechten = createWerklijstRechten(ontkoppeldeDocumentenVerwijderen = true)
+        every {
+            policyService.readWerklijstRechten()
+        } returns werklijstRechten
+        every {
+            ontkoppeldeDocumentenService.find(id)
+        } returns Optional.empty()
+
+        When("the delete endpoint is called with that id") {
+            sut.delete(id)
+
+            Then("there are no exceptions") {
+            }
+        }
+    }
+
+    Given("a document that exists in the database but results in a 404 from OpenZaak") {
+        val werklijstRechten = createWerklijstRechten(ontkoppeldeDocumentenVerwijderen = true)
+        val document = OntkoppeldDocument()
+        document.documentUUID = UUID.randomUUID()
+        document.id = 1
+        every {
+            policyService.readWerklijstRechten()
+        } returns werklijstRechten
+        every {
+            ontkoppeldeDocumentenService.find(document.id)
+        } returns Optional.of(document)
+        every {
+            drcClientService.readEnkelvoudigInformatieobject(document.documentUUID)
+        } throws FoutException(Fout(null, null, null, 404, null, null))
+        every {
+            ontkoppeldeDocumentenService.delete(document.id)
+        } just runs
+
+        When("the delete endpoint is called with the id of that document") {
+            sut.delete(document.id)
+
+            Then("the document is deleted") {
+                verify {
```

Comment:
Do we need to verify this was called only once?
```suggestion
                verify(exactly = 1) {
```